Python: Change note for new points-to implementation.

markshannon opened the pull request:
The missing change note for https://github.com/Semmle/ql/pull/1277 and subsequent related PRs.

felicity-semmle commented on change-notes/1.21/analysis-python.md:
```diff
@@ -3,8 +3,33 @@
+
+* Non-local tracking of bound methods and instances of `super()`
+* Superior analysis of conditionals and thus improved reachability analysis.
+* Superior modelling of descriptors, for example, classmethods and staticmethods.
```
modelling -> modeling

Are "classmethods" and "staticmethods" code elements, or should they each be two words?

markshannon replied:
They are code elements

felicity-semmle commented on change-notes/1.21/analysis-python.md:
```diff
@@ -3,8 +3,33 @@
+
+### Impact on existing queries.
+
+As points-to underpins many queries and provides the call-graph and reachability analysis required for taint-tracking, many queries will have additional results and some may have fewer results.
```
Suggest: "As points-to analysis underpins many queries, and provides the call-graph and reachability analysis required for taint-tracking, the results of many queries will change.

felicity-semmle commented on change-notes/1.21/analysis-python.md:
```diff
@@ -3,8 +3,33 @@
+
+As points-to underpins many queries and provides the call-graph and reachability analysis required for taint-tracking, many queries will have additional results and some may have fewer results.
+
+New results are a result of the improved reachability analysis and non-local tracking of bound-methods.
```
To avoid using "result" in two different senses, perhaps: "The improved reachability analysis and non-local tracking of bound methods will identify new results."?

felicity-semmle commented on change-notes/1.21/analysis-python.md:
```diff
@@ -3,8 +3,33 @@
+
+New results are a result of the improved reachability analysis and non-local tracking of bound-methods.
+Removed results are a result of more precise tracking of values through `*` arguments.
+The expectation is that number of true positives will increase and the number of false negatives will decline.
```
Perhaps: "Overall the number of true positive results should increase and the number false negative results should decline. " ?

felicity-semmle commented on change-notes/1.21/analysis-python.md:
```diff
@@ -3,8 +3,33 @@
+New results are a result of the improved reachability analysis and non-local tracking of bound-methods.
+Removed results are a result of more precise tracking of values through `*` arguments.
+The expectation is that number of true positives will increase and the number of false negatives will decline.
+However, this is new code and may still contain errors.
```
What are we trying to say here? That the API will be developed further, that it's a preliminary version, that we haven't tested it, that we welcome feedback on it...?

markshannon commented:
@felicity-semmle I've addressed your comments. I've used "may" rather than "will" throughout, as we can't find more results that aren't there to be found, nor fix false positives that don't exist.
